Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/Data/AbstractData.cs

```diff
@@ -102,5 +117,17 @@ public JobHandle GetDependencyFor(AccessType accessType)
         {
             return m_AccessController.GetDependencyFor(accessType);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void IncrementVersionIfWrite(AccessType accessType)
+        {
+            if (accessType is AccessType.ExclusiveWrite or AccessType.SharedWrite)
+            {
+                unchecked
```

Comment:
I think it's fine to leave this just to signify that we have thought about it and we want to have the wrap around behaviour but I don't think we need it. By default it should be unchecked.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked

Reply:
You're right that it's not necessary with the default compiler config but if someone turns on overflow checking then we don't want this to trigger when it overflows because it's expected and handled behaviour.

By contrast, `IDProvider` doesn't have an explicit `unchecked` increment because a consumer may want an exception if they don't handle the threshold warnings.

That's my rationale anyway.

Reply:
Yep, that's fair.